client/src/we-shop/filters/filter/filter.tsx

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import InputText from "../../components/input-text/input-text";
+import "./filter.scss";
+import InputRange from "react-input-range";
+import "react-input-range/lib/css/index.css";
+
+import Datepicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import Slider, { Range } from "rc-slider";
+import "rc-slider/assets/index.css";
+import { useDispatch } from "react-redux";
+import { Action } from "redux";
+let x = { min: 2, max: 10 };
+function Filter(props: {
+  filter: { text: string; type: string; value: any; action: CallableFunction };
+}) {
+  const dispatch = useDispatch();
```

Comment:
what is this doing? 
if you need to pass props you can use a pattern on container which is connected to the store and renders the filter component for all the props given in mapStateToProps